lnd.go

```diff
@@ -602,9 +556,9 @@ func genCertPair(certFile, keyFile string) error {
 	return nil
 }
 
-// genMacaroons generates three macaroon files; one admin-level, one
-// for invoice access and one read-only. These can also be used
-// to generate more granular macaroons.
+// genMacaroons generates three macaroon files; one admin-level, one for
+// invoice access and one read-only. These can also be used to generate more
+// granular macaroons.
 func genMacaroons(ctx context.Context, svc *macaroons.Service,
```

Comment:
How to generate an admin macaroon that can access all sub services? In signer, a new entity and action is introduced. I think admin should be able to have permissions to access signer?

Reply:
Is the solution then now to use two macaroons from the client (admin and signer)? I think this also means that two tcp connections need to be created?

Reply:
Is this something that can be controlled client side, or is a change on the server needed?

Reply:
I did a quick try passing multiple macaroon credentials to the dial call, but that is not going to work. `PerRPCCredentials`, didn't try that, but it also means we need to pass around those macaroons. Is it worth the effort at this point? Otherwise I'd say, just make signing possible with the admin macaroon for now.

Reply:
Why do you need to pass multiple macaroon credentials? You should only need to pass the `signer.macaroon` if you're interacting with the signer sub-server. The systems that interact w/ the sub-servers will need to get used to managing their various macaroons, as we can easily implemented domain isolation using them. For example, merchants are already used to using the `invoice.macaroon` everywhere. 

It's possible to create custom macarons that can access _only_ what's needed (so signer and new address for example), as there's an open PR for this, but it isn't yet merged.

Reply:
So I need to open multiple tcp connections then? Swaplet needs signer, walletkit and the main rpc.

Reply:
Possibly, will look into if we can modify the way we manually set the credentials and get back to you!